core/src/main/kotlin/plugability/extensions.kt

```diff
@@ -96,7 +96,8 @@ class ExtendingDSL(private val pluginClass: String, private val extensionName: S
         ExtensionOrdered(extensionPoint, pluginClass, extensionName, action, block)
 
     infix fun <T : Any> Extension<T>.applyIf(condition: DokkaConfiguration.() -> Boolean): Extension<T> =
-        this.setCondition(condition)
+        this.setCondition { this@applyIf.condition(this) && condition(this) }
```

Comment:
IMHO it would be better to just have a `addCondition` or something similar to have a list of conditions. This way you'll just call `conditions.all { it == somepredicate }`  when you need this

Reply:
Changed.